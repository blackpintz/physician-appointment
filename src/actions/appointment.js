const addAppointment = appointment => ({
  type: 'ADD_APPOINTMENT',
  appointment,
});

export default addAppointment;
