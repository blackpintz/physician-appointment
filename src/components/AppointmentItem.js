import React from 'react';
import propTypes from 'prop-types';

const AppointmentItem = ({ item }) => {
  const { city, user, physician } = item;
  const { email } = user;
  const { category } = physician;
  return (
    <>
      <p>Each Appointment goes here.</p>
      <p>{`Email: ${email}`}</p>
      <p>{`Category: ${category}`}</p>
      <p>{city}</p>
    </>
  );
};

AppointmentItem.propTypes = {
  item: propTypes.objectOf(propTypes.oneOfType([
    propTypes.string,
    propTypes.number,
    propTypes.object,
  ])),
};

AppointmentItem.defaultProps = {
  item: { city: 'no value' },
};

export default AppointmentItem;
