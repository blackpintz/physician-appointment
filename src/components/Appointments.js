import React, { useEffect } from 'react';
import axios from 'axios';
import { useHistory } from 'react-router-dom';
import { Button } from 'react-bootstrap';

const Appointments = () => {
  const history = useHistory();
  const currentUser = () => {
    const user = localStorage.getItem('user');
    return (user);
  };

  useEffect(async () => {
    if (localStorage.user) {
      const result = await axios.get('https://frozen-harbor-46293.herokuapp.com/appointments', { headers: JSON.parse(localStorage.user) });
      const { data } = result;
      console.log(data);
    } else {
      console.log('Not logged in yet..');
    }
  }, []);

  const handleSignOut = e => {
    e.preventDefault();
    axios({
      method: 'DELETE',
      url: 'https://frozen-harbor-46293.herokuapp.com/auth/sign_out',
      data: JSON.parse(localStorage.user),
    })
      .then(() => {
        localStorage.removeItem('user');
        history.push('/login');
      });
  };
  return (
    <>
      <h2>Appointments go here!</h2>
      {currentUser() ? (
        <Button className="mt-4" variant="danger" onClick={handleSignOut}>Logout</Button>
      ) : (<></>)}
    </>
  );
};

export default Appointments;
