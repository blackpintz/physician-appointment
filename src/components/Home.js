import React, { useEffect } from 'react';
import axios from 'axios';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import {
  Button, Container, Carousel,
} from 'react-bootstrap';
import addPhysician from '../actions/physician';
import docKit from '../images/stethescope.jpg';

const Home = ({ onFetch, physician }) => {
  useEffect(async () => {
    if (localStorage.user) {
      const result = await axios.get('https://frozen-harbor-46293.herokuapp.com/physicians', { headers: JSON.parse(localStorage.user) });
      const { data } = result;
      onFetch(data);
    } else {
      <p>Not logged in yet..</p>;
    }
  }, []);
  return (
    <Container className="w-50">
      <h3 className="mt-2">Available Physicians</h3>
      <Carousel>
        {physician.map(item => (
          <Carousel.Item key={item.id}>
            <img
              className="d-block w-100"
              src={docKit}
              alt="slide"
            />
            <Carousel.Caption>
              <h3 className="text-dark font-weight-bolder">{item.category}</h3>
              <Link to={`/physician/${item.id}`}>
                <Button variant="success">Book an appointment</Button>
              </Link>
            </Carousel.Caption>
          </Carousel.Item>
        ))}
      </Carousel>
    </Container>
  );
};

const mapStateToProps = state => ({
  physician: state.physician,
});

const mapDispatchToProps = dispatch => ({
  onFetch: data => dispatch(addPhysician(data)),
});

Home.propTypes = {
  onFetch: propTypes.func.isRequired,
  physician: propTypes.arrayOf(propTypes.object),
};

Home.defaultProps = {
  physician: [{ value: '' }],
};

const ConnectHome = connect(mapStateToProps, mapDispatchToProps)(Home);

export { ConnectHome as default, Home };
