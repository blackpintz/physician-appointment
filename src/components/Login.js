import React from 'react';
import { Link } from 'react-router-dom';
import { Form, Button } from 'react-bootstrap';
import axios from 'axios';
import propTypes from 'prop-types';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      error: false,
    };
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  handleLogin = e => {
    const { email, password } = this.state;
    const { routeProps } = this.props;
    const { history } = routeProps;
    e.preventDefault();
    const url = 'https://frozen-harbor-46293.herokuapp.com/auth/sign_in';
    axios({
      url,
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        Accept: 'application/json',
      },
      data: {
        email,
        password,
      },
    }).then(response => {
      localStorage.setItem('user',
        JSON.stringify({
          'access-token': response.headers['access-token'],
          client: response.headers.client,
          uid: response.data.data.uid,
          id: response.data.data.id,
          name: response.data.data.name,
          username: response.data.data.username,
        }));
      this.setState({
        email: '',
        password: '',
        error: true,
      });
      history.push('/');
    }).catch(error => {
      <p>{error}</p>;
      this.setState({ error: true });
    });
  }

  render() {
    const {
      email, password, error,
    } = this.state;
    return (
      <>
        {error && <p>Your login credentials are incorrect.</p>}
        <h3>Log in.</h3>
        <Form onSubmit={this.handleLogin}>
          <Form.Group controlId="formBasicEmail">
            <Form.Label>Email address</Form.Label>
            <Form.Control type="email" placeholder="Enter email" name="email" value={email} onChange={this.handleChange} />
            <Form.Text className="text-muted">
              We will never share your email with anyone else.
            </Form.Text>
          </Form.Group>
          <Form.Group controlId="formBasicPassword">
            <Form.Label>Password</Form.Label>
            <Form.Control type="password" placeholder="Password" name="password" value={password} onChange={this.handleChange} />
          </Form.Group>
          <Button variant="primary" type="submit">
            Submit
          </Button>
        </Form>
        <p>You do not have an account?</p>
        <Link to="/signup">Sign-up</Link>
      </>
    );
  }
}

Login.propTypes = {
  routeProps: propTypes.objectOf(propTypes.object),
  history: propTypes.objectOf(propTypes.func.isRequired),
};

Login.defaultProps = {
  routeProps: { history: 'no value' },
  history: {},
};

export default Login;
