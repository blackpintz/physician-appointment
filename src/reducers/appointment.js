const appointmentReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_APPOINTMENT':
      return action.appointment;
    default:
      return state;
  }
};

export default appointmentReducer;
