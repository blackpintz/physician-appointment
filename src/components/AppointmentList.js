import React, { useEffect } from 'react';
import axios from 'axios';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { Container } from 'react-bootstrap';
import AppointmentItem from './AppointmentItem';
import addAppointment from '../actions/appointment';

const AppointmentList = ({ onFetch, appointment }) => {
  useEffect(async () => {
    if (localStorage.user) {
      const result = await axios.get('https://frozen-harbor-46293.herokuapp.com/appointments', { headers: JSON.parse(localStorage.user) });
      const { data } = result;
      onFetch(data);
    } else {
      <p>Not logged in yet..</p>;
    }
  }, []);

  return (
    <Container>
      <h3>Your Appointments</h3>
      {appointment.length === 0 ? (
        <>
          <h5>You do not have any appointments.</h5>
        </>
      ) : (
        <>
          {appointment.map(item => <AppointmentItem key={item.id} item={item} />)}
        </>
      )}
    </Container>
  );
};

const mapStateToProps = state => ({
  appointment: state.appointment,
});

const mapDispatchToProps = dispatch => ({
  onFetch: data => dispatch(addAppointment(data)),
});

AppointmentList.propTypes = {
  onFetch: propTypes.func.isRequired,
  appointment: propTypes.arrayOf(propTypes.object),
};

AppointmentList.defaultProps = {
  appointment: [{ value: '' }],
};

const ConnectAppointmentList = connect(mapStateToProps, mapDispatchToProps)(AppointmentList);

export { ConnectAppointmentList as default, AppointmentList };
