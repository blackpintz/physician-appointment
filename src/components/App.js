import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';
import '../App.css';
import Login from './Login';
import AppointmentList from './AppointmentList';
import Home from './Home';
import Physician from './Physician';
import currentUser from '../helpers/currentUser';
import Signup from './Signup';
import Menu from './Menu';

const App = () => (
  <>
    <Menu />
    <Switch>
      <Route exact path="/login" render={routeProps => (currentUser() ? <Redirect to="/" /> : <Login routeProps={routeProps} />)} />
      <Route exact path="/">
        {currentUser() ? <Home /> : <Redirect to="/login" />}
      </Route>
      <Route exact path="/signup" render={routeProps => (currentUser() ? <Redirect to="/" /> : <Signup routeProps={routeProps} />)} />
      <Route exact path="/appointments">
        {currentUser() ? <AppointmentList /> : <Redirect to="/login" />}
      </Route>
      <Route exact path="/physician/:id" render={routeProps => (!currentUser() ? <Redirect to="/" /> : <Physician id={routeProps.match.params.id} routeProps={routeProps} />)} />
    </Switch>
  </>
);

export default App;
