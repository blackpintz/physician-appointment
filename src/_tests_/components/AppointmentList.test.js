import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { AppointmentList } from '../../components/AppointmentList';
import appointment from '../fixtures/appointments';

Enzyme.configure({ adapter: new Adapter() });

test('should render AppointmentList component', () => {
  const onFetch = jest.fn();
  const wrapper = shallow(<AppointmentList onFetch={onFetch} appointment={appointment} />);
  expect(wrapper).toMatchSnapshot();
});
