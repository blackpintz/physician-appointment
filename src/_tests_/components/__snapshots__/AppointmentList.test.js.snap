// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppointmentList component 1`] = `
<Container
  fluid={false}
>
  <h3>
    Your Appointments
  </h3>
  <AppointmentItem
    item={
      Object {
        "city": "Mwingi",
        "date": "2021-02-07",
        "id": 1,
        "physician": Object {
          "category": "Eye Specialist",
          "id": 1,
        },
      }
    }
    key="1"
  />
</Container>
`;
